src/dataflow-types/src/types.rs

```diff
@@ -1145,12 +1145,28 @@ pub mod sources {
     }
 
     #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
-    pub struct KafkaSourceConnector {
+    pub enum ConnectorLiteral {
+        Kafka(KafkaSourceConnectorLiteral),
+    }
+
+    #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+    pub struct KafkaSourceConnectorLiteral {
         pub addrs: KafkaAddrs,
-        pub topic: String,
         // Represents options specified by user when creating the source, e.g.
         // security settings.
         pub config_options: BTreeMap<String, String>,
+    }
+
+    #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+    pub enum KafkaSourceConnector {
+        Literal(KafkaSourceConnectorLiteral),
+        Reference(String),
+    }
+
+    #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+    pub struct KafkaSourceInstance {
```

Comment:
What does the term "Instance" here signify?

Reply:
If we make some other cosmetic naming changes, this could remain `KafkaSourceConnector`, I think.

Reply:
The idea is that this struct represents all of the bits required to instantiate a materialized source, a connection is just part of that. I was worried that having too many uses of the `Connector` suffix would make things more confusing, but I'm very amenable to different names.

Reply:
After changing the other name I realized you're right there's no reason to change this at this time.